rest-api/dao/participant_dao.py

```diff
@@ -114,25 +114,49 @@ def _do_update(self, session, obj, existing_obj):
                             else None)
       need_new_summary = True
 
-    if obj.organizationId or obj.siteId or obj.hpoId:
-      site, organization, awardee = self.get_pairing_level(obj)
-      obj.organizationId = organization
-      obj.siteId = site
-      obj.hpoId = awardee
-      if awardee:
-        # get provider link for hpo_id (awardee)
-        obj.providerLink = make_primary_provider_link_for_id(awardee)
+    update_pairing = True
+    if obj.siteId is None and obj.organizationId is None and obj.hpoId is None and \
+      obj.providerLink == 'null':
+      # Prevent unpairing if /PUT is sent with no pairing levels.
+      update_pairing = False
+
+    if update_pairing == True:
+      # site,org,or awardee is sent in request: Get relationships and try to set provider link.
+      if obj.organizationId or obj.siteId or (obj.hpoId >= 0) and (obj.providerLink ==
```

Comment:
Can you add parentheses to indicate order of operations on the and vs. the ors?

Reply:
done. assuming I understood your meaning (which I'm not positive on)